exla/lib/exla/defn.ex

```diff
@@ -733,6 +733,43 @@ defmodule EXLA.Defn do
     )
   end
 
+  defp to_operator(:take_along_axis, [tensor, indices, axis], _ans, state) do
+    indices_shape = op_shape(indices)
+    indices_rank = tuple_size(indices_shape)
+
+    axes_range = 0..(indices_rank-1)
+
+    index_vector_dim = indices_rank
+    slice_sizes = 1 |> List.duplicate(indices_rank)
+    offset_dims = []
+    collapsed_slice_dims = Enum.to_list(axes_range)
+    start_index_map = Enum.to_list(axes_range)
+
+    iotas =
+      Enum.map(axes_range, fn axis ->
+        shape = EXLA.Op.get_shape(indices)
```

Comment:
Not sure if this makes a difference, but this could be extracted and used for every iteration.